src/main/java/jeff/task/TaskList.java

```diff
@@ -0,0 +1,195 @@
+package jeff.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import jeff.exception.FileCorruptException;
+import jeff.exception.JeffException;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks = new ArrayList<>();
```

Comment:
According to the [coding standard](https://se-education.org/guides/conventions/java/index.html#naming), constant names must be all uppercase using underscore to separate words (aka SCREAMING_SNAKE_CASE).

You may want to consider renaming the variable to screaming snake case, or to remove the `final` keyword (I know the Intellj IDE will suggest you to make this variable `final`).